bokehjs/test/unit/document/document.ts

```diff
@@ -123,45 +122,6 @@ class ModelWithConstructTimeChanges extends Model {
 
 Models.register("ModelWithConstructTimeChanges", ModelWithConstructTimeChanges)
 
-namespace ComplicatedModelWithConstructTimeChanges {
-  export type Attrs = p.AttrsOf<Props>
-  export type Props = Model.Props & {
-    list_prop: p.Property<AnotherModel[]>
-    dict_prop: p.Property<{[key: string]: AnotherModel}>
-    obj_prop: p.Property<ModelWithConstructTimeChanges>
-    dict_of_list_prop: p.Property<{[key: string]: Model[]}>
-  }
-}
-
-interface ComplicatedModelWithConstructTimeChanges extends ComplicatedModelWithConstructTimeChanges.Attrs {}
-
-class ComplicatedModelWithConstructTimeChanges extends Model {
-  override properties: ComplicatedModelWithConstructTimeChanges.Props
-
-  constructor(attrs?: Partial<ComplicatedModelWithConstructTimeChanges.Attrs>) {
-    super(attrs)
-  }
-
-  override initialize(): void {
-    super.initialize()
-    this.list_prop = [new AnotherModel()]
-    this.dict_prop = {foo: new AnotherModel()}
-    this.obj_prop = new ModelWithConstructTimeChanges()
-    this.dict_of_list_prop = {foo: [new AnotherModel()]}
-  }
-
-  static {
-    this.define<ComplicatedModelWithConstructTimeChanges.Props>(({Array, Dict, Ref}) => ({
-      list_prop:         [ Array(Ref(AnotherModel)) ],
-      dict_prop:         [ Dict(Ref(AnotherModel)) ],
-      obj_prop:          [ Ref(ModelWithConstructTimeChanges) ],
-      dict_of_list_prop: [ Dict(Array(Ref(Model))) ],
-    }))
-  }
-}
-
-Models.register("ComplicatedModelWithConstructTimeChanges", ComplicatedModelWithConstructTimeChanges)
```

Comment:
Why not preserve a test for the complicated model?

Reply:
I don't see a need for a complex test here, because the purpose of testing this PR is to make sure we can collect change events, whatever they are. A simple tests suffices for this. We don't need to test here whether `HasProps` can detect changes and generate appropriate events. That should be tested in `HasProps`' unit tests, if it's not already.